jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java

```diff
@@ -869,12 +869,44 @@ public List<String> getSoftDeleteProperties(Class<?> clazz) {
      * @return true if entity's {@link EntityEntry} implements {@link EntityEntryHasUuid}
      */
     public boolean hasUuid(MetaClass metaClass) {
-        return metaClass.getAnnotations().containsKey(UUID_KEY_ANN_NAME);
+        String uuidPropertyName = getUuidPropertyName(metaClass);
+        return uuidPropertyName != null;
     }
 
+    /**
+     * Gets name of uuid key property.
+     *
+     * @param clazz entity java class
+     * @return uuid property name or null if it doesn't exist
+     */
     @Nullable
     public String getUuidPropertyName(Class<?> clazz) {
-        return (String) metadata.getClass(clazz).getAnnotations().get(UUID_KEY_ANN_NAME);
+        MetaClass metaClass = metadata.getClass(clazz);
+        return getUuidPropertyName(metaClass);
+    }
+
+    /**
+     * Gets name of uuid key property.
+     *
+     * @param metaClass entity MetaClass
+     * @return uuid property name or null if it doesn't exist
+     */
+    @Nullable
+    public String getUuidPropertyName(MetaClass metaClass) {
+        String uuidProperty = (String) metaClass.getAnnotations().get(UUID_KEY_ANN_NAME);
+        if(uuidProperty != null) {
+            return uuidProperty;
+        } else {
+            MetaClass ancestor = metaClass.getAncestor();
+            while (ancestor != null) {
```

Comment:
As I see there is a `io.jmix.core.metamodel.model.MetaClass#getAncestors` method that returns a full list of metaclass ancestors. No need to do a while loop here if you use this method.

Reply:
Originally done the same way as `getPrimaryKeyName`.
Rewritten with `getAncestors`.